### PR TITLE
PUB-509-Dev added framework for smoke tests

### PR DESCRIPTION
**Pull request by Matthew-Summerbell**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by kurt-uk — APPROVED** [review]

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/header/factory/HeaderFactory.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.futurehearings.pip.smoke.common.header.factory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/header/factory/HeaderFactory.java:8]: Remove unused import

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/header/factory/HeaderFactory.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.http.MediaType;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class HeaderFactory {
+
+    public static Map<String, String> createHeader() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", MediaType.APPLICATION_JSON_VALUE);
+        headers.put("Accept", MediaType.APPLICATION_JSON_VALUE);
+        return headers;
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/header/factory/HeaderFactory.java:18]: Suggest refactor to return  Collections.unmodifiableMap(headers);

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/rest/RestClient.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.futurehearings.pip.smoke.common.rest;
+
+import io.restassured.response.Response;
+
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+
+public class RestClient {
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/rest/RestClient.java:9]: Rename RestClient to RestClientTemplate for file and class names to be consistent with other test types e.g. functional and acceptance tests

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/rest/RestClient.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    public static Response makeGetRequest(final Map<String, String> headers,
+                                          final String authorizationToken,
+                                          final String rootContext) {
+        return given()
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/rest/RestClient.java:14]: Suggest: align the return block code to be consistent with other test types e.g. functional and acceptance tests

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/test/SmokeTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    private Map<String,String> headers = new HashMap<>();
+
+    private String authorizationToken;
+
+    private String rootContext;
+
+    @BeforeAll
+    public void beforeAll(TestInfo info) {
+        log.debug("Test execution Class Initiated: " + info.getTestClass().get().getName());
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/test/SmokeTest.java:60]: Optional.get() method calls without an earlier check that the optional has a value.

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+        Response response = callTokenGeneratorEndpoint(bodyForToken, httpStatus, full_token_apiURL);
+        return response.jsonPath().getString("access_token");
+    }
+
+    public static final Response callTokenGeneratorEndpoint(final String bodyForToken,
+                                                            final HttpStatus badRequest,
+                                                            final String full_token_apiURL) {
+
+        Response response = expect().that().statusCode(badRequest.value())
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:31]: Local variable response is reductant.  Suggest: use coding style consistent with the other test types

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.futurehearings.pip.smoke.common.security;
+
+import static io.restassured.RestAssured.expect;
+
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import lombok.extern.slf4j.Slf4j;
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:7]: Remove unused import

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.http.HttpStatus;
+
+public class OAuthTokenGenerator {
+    public static final String generateOAuthToken(final String token_apiURL,
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:11]: static method declared final

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+        String full_token_apiURL = String.format(token_apiURL, token_apiTenantId);
+        final String bodyForToken = String.format("grant_type=%s&client_id=%s&client_secret=%s&scope=%s",
+                                                  grantType, clientID, clientSecret, scope);
+
+        Response response = callTokenGeneratorEndpoint(bodyForToken, httpStatus, full_token_apiURL);
+        return response.jsonPath().getString("access_token");
+    }
+
+    public static final Response callTokenGeneratorEndpoint(final String bodyForToken,
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:27]: static method declared final

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.http.HttpStatus;
+
+public class OAuthTokenGenerator {
+    public static final String generateOAuthToken(final String token_apiURL,
+                                                  final String token_apiTenantId,
+                                                  final String grantType,
+                                                  final String clientID,
+                                                  final String clientSecret,
+                                                  final String scope,
+                                                  final HttpStatus httpStatus) throws Exception {
```
Comment (ShufenZ) [inline src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:17]: Exception is never thrown in the method

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/rest/RestClient.java (hunk 4 above)
Comment (Matthew-Summerbell) [reply src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/rest/RestClient.java:14]: This is correct? As no switch statement?

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java (hunk 6 above)
Comment (Matthew-Summerbell) [reply src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:31]: Removed file

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java (hunk 7 above)
Comment (Matthew-Summerbell) [reply src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:7]: Removed file

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java (hunk 8 above)
Comment (Matthew-Summerbell) [reply src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:11]: Removed file

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java (hunk 9 above)
Comment (Matthew-Summerbell) [reply src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:27]: Removed file

File: src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java (hunk 10 above)
Comment (Matthew-Summerbell) [reply src/smokeTest/java/uk/gov/hmcts/futurehearings/pip/smoke/common/security/OAuthTokenGenerator.java:17]: Removed file

**Review by kurt-uk — APPROVED** [review]

**Review by ShufenZ — APPROVED** [review]

**Pull request merged**